modal/stub.py

```diff
@@ -290,8 +290,7 @@ async def _run(
                 if mode == StubRunMode.SERVE:
                     # Cancel logs loop since we're going to start another one.
                     logs_loop.cancel()
-                else:
-                    await app.disconnect()
+                await app.disconnect()
```

Comment:
What's the reason for this? I think `stub.serve` intentionally doesn't disconnect the app because the app will be updated in a loop

Reply:
Hm ok maybe I misunderstood the code. I thought the existing code was there because of a race with the singleton closing (see the deleted pre_stop method). Will take a second look at it.